src/main/java/org/apache/accumulo/examples/cli/ClientOpts.java

```diff
@@ -0,0 +1,143 @@
+    @Override
+    public ColumnVisibility convert(String value) {
+      return new ColumnVisibility(value);
+    }
+  }
+
+  public static class TimeConverter implements IStringConverter<Long> {
+    @Override
+    public Long convert(String value) {
+      if(value.matches("[0-9]+"))
+        value = "PT"+value+"S"; //if only numbers then assume seconds
+      return Duration.parse(value).toMillis();
+    }
+  }
+
+  public static class MemoryConverter implements IStringConverter<Long> {
+    @Override
+    public Long convert(String str) {
+      try {
+        char lastChar = str.charAt(str.length() - 1);
+        int multiplier = 0;
+        switch (Character.toUpperCase(lastChar)) {
+          case 'G':
+            multiplier += 10;
+          case 'M':
+            multiplier += 10;
+          case 'K':
+            multiplier += 10;
+          case 'B':
+            break;
+          default:
+            return Long.parseLong(str);
+        }
+        return Long.parseLong(str.substring(0, str.length() - 1)) << multiplier;
+      } catch (Exception ex) {
+        throw new IllegalArgumentException("The value '" + str + "' is not a valid memory setting. A valid value would a number "
+            + "possibily followed by an optional 'G', 'M', 'K', or 'B'.");
+      }
+    }
+  }
+
+  public static class PropertiesConverter implements IStringConverter<Configuration> {
+    @Override
+    public Configuration convert(String filename) {
+      try {
+        return new PropertiesConfiguration(filename);
+      } catch (ConfigurationException e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  @Parameter(names = {"-c", "--conf"}, required = true, converter = PropertiesConverter.class,
+      description = "Config file for connecting to Accumulo.  See README.md for details.")
+  private Configuration config = null;
+
+  @Parameter(names = {"-auths", "--auths"}, converter = AuthConverter.class, description = "the authorizations to use when reading or writing")
+  public Authorizations auths = Authorizations.EMPTY;
+
+  public Connector getConnector() {
+    try {
+      ZooKeeperInstance zki = new ZooKeeperInstance(config);
+      return zki.getConnector(getPrincipal(), getToken());
+    } catch (AccumuloException | AccumuloSecurityException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public ClientConfiguration getClientConfiguration() {
+    return new ClientConfiguration(config);
+  }
+
+  public String getPrincipal() {
+    return config.getString("accumulo.examples.principal", "root");
+  }
+
+  public AuthenticationToken getToken() {
+    if (config.containsKey("instance.rpc.sasl.enabled")) {
+      try {
+        if (config.containsKey("accumulo.examples.keytab")) {
+          String keytab = config.getString("accumulo.examples.keytab");
+          return new KerberosToken(getPrincipal(), new File(keytab));
+        } else {
+          return new KerberosToken(getPrincipal());
```

Comment:
The `KerberosToken(String)` constructor expects that the user is already logged in via the JAAS APIs or Hadoop's UserGroupInformation (which is sugar on top of JAAS APIs). As a user, it would be common to use a ticket-cache instead of a keytab to run an example.

UGI does have a method for acquiring an instance from a ticket-cache, `UserGroupInformation.getUGIFromTicketCache(String ticketCache, String user)`; however, this only returns the UGI instance and does not update the static state inside of UGI after this method is called (like `loginUserFromKeytab(..)` does).

I can see two ways forward:
1. Try to use `UGI.loginUserFromSubject(Subject)` after performing the `UserGroupInformation.getUGIFromTicketCache(..)` call. I think this would work, but I'm not 100% positive.
2. Call `UserGroupInformation.getUGIFromTicketCache(..)` at the beginning of each example and wrap all future execution of the example in a `doAs(..)` (instance method on the UGI object returned by `getUGIFromTicketCache`). This is a bit more invasive, but logins are typically one of the first things that could/should happen with Kerberos.